### DISCUSSION TITLE
Return immediately and then do work after the fact?

### DISCUSSION THREAD

**Discussion by tobymurray** [rwf2/Rocket#2009] [Questions]
My goal is to send an email containing the body of a request. I want to return a response immediately, then send the email out of band of the request/response flow.

Is there an idiomatic way to return first then do work after the fact (accessing the request body, in particular)?

**Comment by admwrd** [comment]
Have you looked into something like RabbitMQ or Kafka?

It sounds like there can be a separation of concerns. The first would be adding something to your Queue, and returning OK if it was successfully added. The second would be actually processing through that Queue, where one consumer can be to find emails that need to be sent, for example.

**Reply by somehowchris** [reply]
YES finally a worthy engineer on these threads 😆 Talking from the bottom of my heart. Go for engineering principles and not stuffing everything all together like the other thread suggested

Indeed, would go with rmq since he‘s talking about queuing and not pub-sub like things. One easy implementation would be with `rust-celery`(https://github.com/rusty-celery/rusty-celery)
